YSMOD-650

lurajon opened the pull request:
- vis dokumenter for utgående og innkommende journalposter
- utvidet test

lurajon committed:
YSMOD-650

- vis dokumenter for utgående og innkommende journalposter
- utvidet test

lurajon committed:
YSMOD-650

- fikset litt kotlin syntaks

olsnerik commented on src/main/kotlin/no/nav/yrkesskade/saksbehandling/service/BehandlingService.kt:
```diff
@@ -66,17 +67,14 @@ class BehandlingService(
+        var utgaaendeDokumenter = emptyList<DokumentInfo>()
+        // dersom vi har inngående og utgående journalpost, må vi hente alle dokumenter
+        if (behandling.utgaaendeJournalpostId != null) {
+            val journalpostResult = safClient.hentOppdatertJournalpost(behandling.journalpostId)
```
```suggestion
            val journalpostResult = safClient.hentOppdatertJournalpost(behandling.utgaaendeJournalpostId)
```

olsnerik reviewed (changes requested):
Tror jeg fant en liten bug :) Vet ikke om det i så fall gjør at testen bør styrkes? Ellers fint!

lurajon committed:
Update src/main/kotlin/no/nav/yrkesskade/saksbehandling/service/BehandlingService.kt

Co-authored-by: Erik Prøis Olsnes <90706895+olsnerik@users.noreply.github.com>

lurajon committed:
YSMOD-650

- laget en JournalpostFactory for å lage bedre testdata
- setter utgående journalpost id til Detaljertbehandling DTO
- fikset tester

olsnerik commented on src/main/kotlin/no/nav/yrkesskade/saksbehandling/service/BehandlingService.kt:
```diff
@@ -396,4 +394,17 @@ class BehandlingService(
+            val journalpost = journalpostResult.journalpost
+            journalpost.dokumenter!!.map {
+                val dokument = it!!
+                val journalstatus = if (journalpost.journalstatus != null) journalpost.journalstatus.name  else "Status ikke satt"
+                val journalposttype = if (journalpost.journalposttype != null) journalpost.journalposttype.name else "Type ikke satt"
```
Bare et forslag for å gjøre det litt penere her
```suggestion
                val journalstatus = journalpost.journalstatus?.name ?: "Status ikke satt"
                val journalposttype = journalpost.journalposttype?.name ?: "Type ikke satt"
```

lurajon committed:
Update src/main/kotlin/no/nav/yrkesskade/saksbehandling/service/BehandlingService.kt

Co-authored-by: Erik Prøis Olsnes <90706895+olsnerik@users.noreply.github.com>
